src/OSPSuite.Core/Domain/Services/ConcurrencyManager.cs

```diff
@@ -23,17 +23,20 @@ public interface IConcurrencyManager
 
    public class ConcurrencyManager : IConcurrencyManager
    {
-      private readonly ICoreUserSettings _coreUserSettings;
-
+      //private readonly ICoreUserSettings _coreUserSettings;
+      private int _maximumNumberOfCoresToUse = 2;
+      //TODO: Check with Michael why ICoreUserSettings is not available through container on Core
+      /*
       public ConcurrencyManager(ICoreUserSettings coreUserSettings)
       {
          _coreUserSettings = coreUserSettings;
       }
-
+      */
       public async Task<IReadOnlyDictionary<TData, TResult>> RunAsync<TData, TResult>(int numberOfCoresToUse, CancellationToken cancellationToken, IReadOnlyList<TData> data, Func<int, CancellationToken, TData, Task<TResult>> action)
       {
          if (numberOfCoresToUse <= 0)
```

Comment:
That should never happen? Or in this case, the numberOCoresToUse is the Max -1?

Reply:
This is passed by the user. This way the user has control on its resources. You could have 8 cores but would like to use only 6 because you are doing something more. If you pass 0 or below it is interpreted as take all available.

Reply:
we always take AllAvailable -1 becuase otherwise your computer just freezes completely
So I would say we should stuck with this. This is also a constant in ospsuiteREnv